envoy/stream_info/stream_info.h

```diff
@@ -205,6 +205,16 @@ struct ResponseCodeDetailValues {
 
 using ResponseCodeDetails = ConstSingleton<ResponseCodeDetailValues>;
 
+struct StreamStateStringValues {
+  const std::string StreamStarted = "Started";
+  const std::string StreamInProgress = "InProgress";
+  const std::string StreamEnded = "Ended";
+};
+
+using StreamStateStrings = ConstSingleton<StreamStateStringValues>;
+
+enum class StreamState { Started, InProgress, Ended };
```

Comment:
I wonder if we may want this to be a set of flags -- i.e. a stream can be both Started and InProgress. This will allow us to add additional state in the future such as RequestHeadersRead, RequestComplete, ResponseHeadersRead, etc.

Reply:
That sounds like a good idea, see response code flags for an example.